### PR TITLE
Edited DBC file & changed BMSW CAN function parameters

### PR DESCRIPTION
**Pull request by Gray-man**
Edited the DBC file & CAN message functions so that the messages are packed correctly. Also edited the CAN message function arguments so that no data is lost from the implicit conversion

### CODE REVIEW

File: components/bms_worker/src/CAN/BMS_msgTx.h [hunk 1]
```diff
@@ -69,279 +69,279 @@ static inline void set_Counter(CAN_data_T* message, uint8_t value)
...
 static inline void set_CoolPercentage0(CAN_data_T* message, uint8_t value)
 {
-    message->u64 |= ((uint64_t)(value & 0x7F)) << 56;   // 7 bits, 1% precision, range [0-128]%
+    message->u64 |= ((uint64_t)(value & 0x7F)) << 22;   // 7 bits, 1% precision, range [0-128]%
 }
 
 static inline void set_CoolState1(CAN_data_T* message, bool value)
 {
-    message->u64 |= ((uint64_t)((value ? 0x01 : 0U))) << 55;
+    message->u64 |= ((uint64_t)((value ? 0x01 : 0U))) << 21; // I'm not sure if this is a 1 bit or 1 byte value. If it's 1 bit, this message can be 4 bytes instead of 6. Same with CoolState0
```
Comment (rickyelopez) [inline components/bms_worker/src/CAN/BMS_msgTx.h:331]: 0x01 is 1 bit, so you're packing 1 bit here.

**Review by rickyelopez — COMMENTED** [review]
Did you intend to update a bunch of submodules?

IMO your time would be better spent working on the autogenerator rather than manually generating this stuff. See my WIP [here](https://github.com/concordia-fsae/firmware/pull/47).

Otherwise, seems fine.

**Comment by JoshLafleur** [comment]
> Did you intend to update a bunch of submodules?
> 
> IMO your time would be better spent working on the autogenerator rather than manually generating this stuff. See my WIP [here](https://github.com/concordia-fsae/firmware/pull/47).
> 
> Otherwise, seems fine.

Yea submodules should not be updated, although commit/branch is also not tracked for the most part. Should probably be done to prevent platform diversion long term. Definetly agree autogenerator should be a long term focus, this is a short term fix to not having it implemented presently

**Comment by JoshLafleur** [comment]
@Gray-man Commit message is "Combined last 2 commits into 1", this does not detail what the commit chages. The commit message should be what the changes in this PR detail

**Comment by rickyelopez** [comment]
> although commit/branch is also not tracked for the most part

what do you mean by this?

**Comment by JoshLafleur** [comment]
> > although commit/branch is also not tracked for the most part
> 
> 
> 
> what do you mean by this?

So clone the repo (or i guess fork it) into a repo on concorddia, then submodule off that and only update at specific points after testing the new library

**Comment by rickyelopez** [comment]
> > > although commit/branch is also not tracked for the most part
> > 
> > 
> > what do you mean by this?
> 
> So clone the repo (or i guess fork it) into a repo on concorddia, then submodule off that and only update at specific points after testing the new library

Why? What does that give us over just using hashes from the existing upstream repo? Especially if we don't actually plan to modify anything internally. If all we're doing is pulling in a dependency, the only thing cloning the repo buys us is protection from the upstream repo being deleted altogether.

**Comment by JoshLafleur** [comment]
> > > > although commit/branch is also not tracked for the most part
> 
> > > 
> 
> > > 
> 
> > > what do you mean by this?
> 
> > 
> 
> > So clone the repo (or i guess fork it) into a repo on concorddia, then submodule off that and only update at specific points after testing the new library
> 
> 
> 
> Why? What does that give us over just using hashes from the existing upstream repo? Especially if we don't actually plan to modify anything internally. If all we're doing is pulling in a dependency, the only thing cloning the repo buys us is protection from the upstream repo being deleted altogether.

I had thought we could do that but when i was just looking at the gitmodules documentation i couldnt find info on specifying a commit hash. Fundamentally the purpose is the same and yes using their hash ultimately is the bestfor our needs/uses. Do you have a gitmodules example of a specified hash? Im on my phone and couldnt find one

**Comment by rickyelopez** [comment]
> > > > > although commit/branch is also not tracked for the most part
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > what do you mean by this?
> > 
> > 
> > > 
> > 
> > 
> > > So clone the repo (or i guess fork it) into a repo on concorddia, then submodule off that and only update at specific points after testing the new library
> > 
> > 
> > Why? What does that give us over just using hashes from the existing upstream repo? Especially if we don't actually plan to modify anything internally. If all we're doing is pulling in a dependency, the only thing cloning the repo buys us is protection from the upstream repo being deleted altogether.
> 
> I had thought we could do that but when i was just looking at the gitmodules documentation i couldnt find info on specifying a commit hash. Fundamentally the purpose is the same and yes using their hash ultimately is the bestfor our needs/uses. Do you have a gitmodules example of a specified hash? Im on my phone and couldnt find one

Submodule hashes are tracked. That's exactly the change that @Gray-man made here that he needed to revert, he accidentally checked in a different hash of a bunch of submodules. The hash doesn't explicitly show up in the `.gitmodules` file, but git is tracking the hash, and the specific hash that is tracked is the one that gets cloned when you tell it to fetch the submodules.

**Comment by JoshLafleur** [comment]
> > > > > > although commit/branch is also not tracked for the most part
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > >
> 
> > > 
> 
> > > 
> 
> > > > > what do you mean by this?
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > So clone the repo (or i guess fork it) into a repo on concorddia, then submodule off that and only update at specific points after testing the new library
> 
> > > 
> 
> > > 
> 
> > > Why? What does that give us over just using hashes from the existing upstream repo? Especially if we don't actually plan to modify anything internally. If all we're doing is pulling in a dependency, the only thing cloning the repo buys us is protection from the upstream repo being deleted altogether.
> 
> > 
> 
> > I had thought we could do that but when i was just looking at the gitmodules documentation i couldnt find info on specifying a commit hash. Fundamentally the purpose is the same and yes using their hash ultimately is the bestfor our needs/uses. Do you have a gitmodules example of a specified hash? Im on my phone and couldnt find one
> 
> 
> 
> Submodule hashes are tracked. That's exactly the change that @Gray-man made here that he needed to revert, he accidentally checked in a different hash of a bunch of submodules. The hash doesn't explicitly show up in the `.gitmodules` file, but git is tracking the hash, and the specific hash that is tracked is the one that gets cloned when you tell it to fetch the submodules.

Im at the bounds of what I know for submodules tbh. What would be the way to revert this? For both me and lucas to learn and fix this pr

**Comment by rickyelopez** [comment]
> > > > > > > although commit/branch is also not tracked for the most part
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > > >
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > > >
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > > > what do you mean by this?
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > >
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > > So clone the repo (or i guess fork it) into a repo on concorddia, then submodule off that and only update at specific points after testing the new library
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > Why? What does that give us over just using hashes from the existing upstream repo? Especially if we don't actually plan to modify anything internally. If all we're doing is pulling in a dependency, the only thing cloning the repo buys us is protection from the upstream repo being deleted altogether.
> > 
> > 
> > > 
> > 
> > 
> > > I had thought we could do that but when i was just looking at the gitmodules documentation i couldnt find info on specifying a commit hash. Fundamentally the purpose is the same and yes using their hash ultimately is the bestfor our needs/uses. Do you have a gitmodules example of a specified hash? Im on my phone and couldnt find one
> > 
> > 
> > Submodule hashes are tracked. That's exactly the change that @Gray-man made here that he needed to revert, he accidentally checked in a different hash of a bunch of submodules. The hash doesn't explicitly show up in the `.gitmodules` file, but git is tracking the hash, and the specific hash that is tracked is the one that gets cloned when you tell it to fetch the submodules.
> 
> Im at the bounds of what I know for submodules tbh.

That's OK, that's why I'm explaining it


> What would be the way to revert this? For both me and lucas to learn and fix this pr

I would do a `git reset HEAD^` which will "undo" the top commit (drop it but keep all the changes as unstaged), then you can `git add <specific file>` all of your _actual_ changes that you want (don't `git add` any of the submodule dirs, even though they will show up as unstaged changes). Then you can commit and push again as usual.

**Comment by rickyelopez** [comment]
let me know if you need help with this

**Pull request closed without merging**